### PR TITLE
Refactoring and renaming of HoloCube and interfaces

### PR DESCRIPTION
**Pull request by philippjfr**
This PR includes renaming of Cube to HoloCube updating tests and documentation to match. It also includes some reorganization of the cube module including the creation of a util module.

### CODE REVIEW

**Comment by philippjfr** [comment]
Would be good to merge this asap as it includes important renaming and fixes/updates to the prototype notebook.

File: holocube/element/util.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from iris.util import guess_coord_axis
```
Comment (jlstevens) [inline holocube/element/util.py:1]: A utility file does make sense at this point.

File: holocube/element/cube.py [hunk 2]
```diff
@@ -96,7 +69,15 @@ def points(self, kdims=None, vdims=None, mdims=None, **kwargs):
 
 
 
-class CubeData(DataColumns):
+class CubeData(GridColumns):
```
Comment (jlstevens) [inline holocube/element/cube.py]: I see the super class is now `GridColumns` which makes sense (it defines the interface for gridded data like this). Not sure about the name `CubeData` although I'm not certain we can fit it into the `*Columns` naming scheme either...

**Edit:** How about `CubeInterface`?

File: holocube/element/cube.py [hunk 3]
```diff
@@ -127,56 +110,53 @@ def init(cls, eltype, data, kdims, vdims):
...
 
     @classmethod
-    def validate(cls, columns):
+    def validate(cls, holocube):
         pass
 
 
     @classmethod
-    def values(cls, columns, dim, expanded=True, flat=True):
+    def values(cls, holocube, dim, expanded=True, flat=True):
```
Comment (jlstevens) [inline holocube/element/cube.py:118]: I might have left the name as `columns` though I don't mind `holocube` if you think it is clearer...

**Comment by jlstevens** [comment]
Other than a few minor comments above, it all looks good and I think it is ready to merge once I've seen your replies.

**Comment by jlstevens** [comment]
Ok, the tests have just passed and I understand it is ready to merge.

**Pull request merged**